tests/test_token_gen.py

```diff
@@ -212,20 +244,30 @@ def test_invalid_params(self, auth_app, values):
             auth.create_custom_token(user, claims, app=auth_app)
 
     def test_noncert_credential(self, user_mgt_app):
+        if _is_emulated():
+            # Should work fine with the emulator, so do a condensed version of
+            # test_sign_with_iam below.
+            custom_token = auth.create_custom_token(MOCK_UID, app=user_mgt_app).decode()
+            self._verify_signer(custom_token, _token_gen.AUTH_EMULATOR_EMAIL)
+            return
         with pytest.raises(ValueError):
             auth.create_custom_token(MOCK_UID, app=user_mgt_app)
 
     def test_sign_with_iam(self):
-        options = {'serviceAccountId': 'test-service-account', 'projectId': 'mock-project-id'}
+        signer = _token_gen.AUTH_EMULATOR_EMAIL if _is_emulated() else 'test-service-account'
```

Comment:
This test doesn't use any of the fixtures. Does it ever run in the emulator mode?

Update: I see a few other tests in this file that doesn't depend on the fixtures, and therefore never runs in the emulator mode. Let's not modify them.

Reply:
It might not use those fixtures, but the presence of those fixtures probably affects it via the environment variable. At any rate, I did get failures for this and the tests below, where the email address in the assertion failure definitely was the one used by the emulator code.

Reply:
That sounds like a bug to me. With function-scoped fixtures, these tests should never see the emulator mode enabled.

Reply:
Ah, yes, with the change to function-scoped fixtures, they no longer see emulator mode enabled.